src/main/java/org/embulk/parser/jsonpath/JsonpathParserPlugin.java

```diff
@@ -172,37 +172,19 @@ public void run(TaskSource taskSource, Schema schema,
                     continue;
                 }
 
-                if (!json.isArray()) {
-                    skipOrThrow(new JsonRecordValidateException(format(Locale.ENGLISH,
-                            "Json string is not representing array value json='%s'", json)), stopOnInvalidRecord);
-                    continue;
-                }
-
-                for (JsonNode recordValue : json) {
-                    try {
-                        if (recordValue.getNodeType() != JsonNodeType.OBJECT) {
-                            throw new JsonRecordValidateException(format(Locale.ENGLISH,
-                                    "Json string is not representing map value json='%s'", recordValue));
+                if (json.isArray()) {
+                    for (JsonNode recordValue : json) {
+                        try {
+                            createRecordFromJson(recordValue, schema, jsonPathMap, visitor, pageBuilder);
                         }
-
-                        for (Column column : schema.getColumns()) {
-                            JsonNode value = null;
-                            if (jsonPathMap.containsKey(column)) {
-                                try {
-                                    value = JsonPath.using(JSON_PATH_CONFIG).parse(recordValue).read(jsonPathMap.get(column));
-                                }
-                                catch (PathNotFoundException e) {
-                                    // pass (value is nullable)
-                                }
-                            }
-                            else {
-                                value = recordValue.get(column.getName());
-                            }
-                            visitor.setValue(value);
-                            column.visit(visitor);
+                        catch (DataException e) {
+                            skipOrThrow(e, stopOnInvalidRecord);
+                            continue;
                         }
-
-                        pageBuilder.addRecord();
+                    }
+                } else {
```

Comment:
Could you fix the style error?

```
[ant:checkstyle] [ERROR] .../src/main/java/org/embulk/parser/jsonpath/JsonpathParserPlugin.java:185:17: '}' at column 17 should be alone on a line. [RightCurly]
```

```suggestion
                }
                else {
```